storagenode/storagenodedb/consoledb.go

```diff
@@ -54,3 +57,152 @@ func (db *consoledb) GetSatelliteIDs(ctx context.Context, from, to time.Time) (_
 
 	return satellites, nil
 }
+
+// GetDailyBandwidthUsed returns slice of daily bandwidth usage for provided time range,
+// sorted in ascending order
+func (db *consoledb) GetDailyTotalBandwidthUsed(ctx context.Context, from, to time.Time) (_ []console.BandwidthUsed, err error) {
+	defer mon.Task()(&ctx)(&err)
+	defer db.locked()()
+
+	since, _ := getDateEdges(from.UTC())
+	_, before := getDateEdges(to.UTC())
+
+	rows, err := db.db.QueryContext(ctx, db.Rebind(`
+		SELECT action, SUM(amount), created_at
+		FROM bandwidth_usage
+		WHERE ? <= created_at AND created_at <= ?
+		GROUP BY DATE(created_at), action
+		ORDER BY created_at ASC`), since, before)
+	if err != nil {
+		return nil, err
+	}
+
+	defer func() {
+		err = errs.Combine(err, rows.Close())
+	}()
+
+	var dates []time.Time
+	dailyBandwidth := make(map[time.Time]*console.BandwidthUsed, 0)
+
+	for rows.Next() {
+		var action int32
+		var amount int64
+		var createdAt time.Time
+
+		err = rows.Scan(&action, &amount, &createdAt)
+		if err != nil {
+			return nil, err
+		}
+
+		from, to := getDateEdges(createdAt)
+
+		bandwidthUsed, ok := dailyBandwidth[from]
+		if !ok {
+			bandwidthUsed = &console.BandwidthUsed{
+				From: from,
+				To:   to,
+			}
+
+			dates = append(dates, from)
+			dailyBandwidth[from] = bandwidthUsed
+		}
+
+		switch pb.PieceAction(action) {
+		case pb.PieceAction_GET:
+			bandwidthUsed.Egress.Usage = amount
+		case pb.PieceAction_GET_AUDIT:
+			bandwidthUsed.Egress.Audit = amount
+		case pb.PieceAction_GET_REPAIR:
+			bandwidthUsed.Egress.Repair = amount
+		case pb.PieceAction_PUT:
+			bandwidthUsed.Ingress.Usage = amount
+		case pb.PieceAction_PUT_REPAIR:
+			bandwidthUsed.Ingress.Repair = amount
+		}
+	}
+
+	var bandwidthUsedList []console.BandwidthUsed
+	for _, date := range dates {
+		bandwidthUsedList = append(bandwidthUsedList, *dailyBandwidth[date])
+	}
+
+	return bandwidthUsedList, nil
+}
+
+// GetDailyBandwidthUsed returns slice of daily bandwidth usage for provided time range,
+// sorted in ascending order for particular satellite
+func (db *consoledb) GetDailyBandwidthUsed(ctx context.Context, satelliteID storj.NodeID, from, to time.Time) (_ []console.BandwidthUsed, err error) {
```

Comment:
is there a way we can combine the logic between these two functions some? there is a lot of overlap. a helper method perhaps?

Reply:
Yeah, I agree just trying to figure how take out some independent responsibility from those methods, so it won't result in a scenario when I need to reimplement 3 functions instead of one, if behavior of one of those methods should be changed